File: data/python/BINARY_SEARCH_TRANSLATED.java
====================
public static int fGold ( int [ ] arr , int l , int r , int x ) {
  if ( r >= l ) {
    int mid = l + ( r - l ) / 2 ;
    if ( arr [ mid ] == x ) return mid ;
    else if ( arr [ mid ] > x ) return fGold ( arr , l , mid - 1 , x ) ;
    else return fGold ( arr , mid + 1 , r , x ) ;
  }
  else return - 1 ;
}

File: data/java/BINARY_SEARCH_TRANSLATED.py
====================
def pythree_search ( arr , l , r , x ) :
    if r >= l :
        mid = l + ( r - l ) // 2
        if arr [ mid ] == x :
            return mid
        if arr [ mid ] > x :
            return pythree_search ( arr , l , mid - 1 , x )
        return pythree_search ( arr , mid + 1 , r , x )
    return - 1

